threads/flask_api.py

```diff
@@ -71,11 +71,16 @@ def forward_tick():
             if self.check_duplicate(tick):
                 return "duplicate request please wait 10s", 400
 
-            if not validate_tick(tick, self.clockchain.latest_selected_tick(),
-                                 self.clockchain.possible_previous_ticks()):
-                return "Invalid tick", 400
+            tick_is_valid, reason_is_resync_trigger = validate_tick(tick, self.clockchain.latest_selected_tick(),
```

Comment:
Cannot use this construction, it breaks timeminer.py here: 

if not validate_tick(tick, prev_tick, possible_previous, verbose=False):

_(breaks by doing if not (Bool, Bool) which has unintended consequences)_

The validate_XYZ functions all only return 1 Boolean, let's keep it that way otherwise the usage will be very confusing 

Also a resync check is probably not needed for future code. For example an implementation of validate_clockchain would use validate_tick as a subroutine, but there you are not at all interested in checking for resync flags.. 

An alternative construction you could use is to pass a boolean to the function as follows: 

validate_tick(....., ....., ....., resync_check=True) where the default is False in the function

Then in the function itself, you only return (Bool, Bool) in relevant place if resync check is true, Otherwise you return only 1 Bool as it was before

Reply:
Updated the code before you edited your comment. Your solution would involve a bunch of if statements and duplication of code just to return a different format. My current solution is ugly. There must be a better solution?